runner/debug/debug.go

```diff
@@ -0,0 +1,75 @@
+package debug
+
+import (
+	"fmt"
+)
+
+// productFilterString takes a product name and a slice of filter strings, and produces valid, product-specific filter flags.
+// The returned string is in the form " -target=metrics -target=pprof" (for Vault), " -capture=host" (for Consul), or " -event-topic=Allocation" (for Nomad)
+func productFilterString(product string, filters []string) (string, error) {
+	var filterString string
+	var legalFilters map[string]bool
+	var optFlag string
+
+	// Define valid filter flagnames and values for all products
+	nomadOptFlag := "event-topic"
+	nomadFilters := map[string]bool{
+		"*":          true,
+		"ACLToken":   true,
+		"ACLPolicy":  true,
+		"ACLRole":    true,
+		"Job":        true,
+		"Allocation": true,
+		"Deployment": true,
+		"Evaluation": true,
+		"Node":       true,
+		"Service":    true,
+	}
+
+	vaultOptFlag := "target"
+	vaultFilters := map[string]bool{
```

Comment:
Do we know how often these change across products? I worry that we could end up tightly coupled to product versions.

Reply:
I'm not sure, but I don't think it's frequent. Is it OK if I investigate this as part of the next two runners I build (Nomad and Consul) -- In general I'm happy to make this a simple passthrough and not try to track/enforce valid filter options for the underlying debug command.

Reply:
Yeah, I think it's fine to investigate in the upcoming work items.